great_expectations/rule_based_profiler/domain_builder/domain.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import IDDict
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.types import SerializableDictDot
+
+
+@dataclass
+class Domain(SerializableDictDot):
+    domain_kwargs: Optional[Dict[str, Union[str, Dict[str, Any]]]] = None
```

Comment:
when is the value of a domain_kwarg a dictionary?

Reply:
@jcampbell I thought that we might want this for the future, but cannot find usecases right now, so removed (also, importantly, please see below that we are now making `Domain` extend `IDDict`.  Thanks.